protocol/x/prices/keeper/slinky_adapter.go

```diff
@@ -66,13 +62,42 @@ func (k Keeper) GetIDForCurrencyPair(ctx sdk.Context, cp slinkytypes.CurrencyPai
 
 		// compare the currency pairs to the one that we're looking for
 		if strings.EqualFold(mpCp.String(), cp.String()) {
+			k.AddCurrencyPairIDToStore(ctx, mp.Id, cp)
+
 			return uint64(mp.Id), true
 		}
 	}
 
 	return 0, false
 }
 
+func (k Keeper) GetCurrencyPairIDFromStore(ctx sdk.Context, cp slinkytypes.CurrencyPair) (marketId uint64, found bool) {
```

Comment:
Can we add some basic unit tests for below functions?

Reply:
done!